src/dns.rs

```diff
@@ -61,14 +61,29 @@ impl ToIpAddr for IpAddr {
     }
 }
 
+impl ToSocketAddr for (String, u16) {
```

Comment:
A comment saying we expect the u16 to be a port would be helpful, even if obvious.

Reply:
done